base/memory/castanets_memory_mapping.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 Samsung Electronics. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef BASE_MEMORY_CASTANETS_MEMORY_MAPPING_H_
+#define BASE_MEMORY_CASTANETS_MEMORY_MAPPING_H_
+
+#include <vector>
+
+#include "base/unguessable_token.h"
+#include "base/memory/ref_counted.h"
+
+namespace base {
+
+class BASE_EXPORT CastanetsMemoryMapping
+    : public RefCountedThreadSafe<CastanetsMemoryMapping> {
+ public:
+  static scoped_refptr<CastanetsMemoryMapping> Create(
+      const UnguessableToken& id, size_t size);
+
+  void AddMapping(void* address);
+  void RemoveMapping(void* address);
+
+  UnguessableToken guid() const { return guid_; }
+  size_t mapped_size() const { return mapped_size_; }
+  void* GetMemory();
+
+  bool HasMapping() const { return !addresses_.empty(); }
+
+  void* MapForSync(int fd);
+  void UnmapForSync(void* memory);
+
+ private:
+  friend class base::RefCountedThreadSafe<CastanetsMemoryMapping>;
+
+  CastanetsMemoryMapping(const UnguessableToken& id, size_t size);
+  ~CastanetsMemoryMapping();
+
+  UnguessableToken guid_;
+  size_t mapped_size_ = 0;
+
+  std::vector<void*> addresses_;
```

Comment:
I think each instance of CastanetsMemoryMapping is unique information, so how about making it uncopyable and unassignable? We can do it using DISALLOW_COPY_AND_ASSIGN macro.

Reply:
Good point. I've added the macro.